mgr/src/server/actions.cr

```diff
@@ -36,13 +38,17 @@ module Action
   end
 
   def self.dispatch(name : String, pool_name : String, nodes : Array(MoanaTypes::Node), data : String)
-    # TODO Set token
     headers = HTTP::Headers.new
     headers["Content-Type"] = "application/json"
     resp = Response.new
     # TODO: Send requests concurrently and handle
     # the response
+
+    # Add tokens to nodes[execpt for new additions] for authorization.
+    Datastore.node_tokens(nodes)
```

Comment:
`nodes = Datastore.node_tokens(nodes)`

Reply:
Ack. I thought it would update internally.

Reply:
It is updating since `MoanaTypes::Node` is a class. But good to keep datastore apis as return by value to avoid bugs later.

Reply:
cool!